### PR TITLE
Add User enable/disable pairing endpoint

### LINKED ISSUE #34: Add user endpoint to disable/enable pairing
**Describe the task that needs to be done.**
We need an endpoint to disable/enable pairing for a user so that a user can opt in to being paired with a buddy.

**Describe how a solution to your proposed task might look like (and any alternatives considered).**
Add a `pairingEnabled` boolean field to the User entity

Options for endpoint:
1. Update `pairingEnabled` field through an endpoint which accepts a UserDTO in the request body
PUT api/users/{id}
Request:
Path param: User id
Request body: UserDTO
Response:
200 OK w/ updated User entity
404 Not Found

Pros:
- Can be reused for any other fields that may need to be updated

Cons:
- Need check for whether any fields are left out of the UserDTO, and not update those fields in the database. If  `Pair` in the DTO is null, how do we determine whether the pair is being intentionally set to null to remove the pair?

2. Update `pairingEnabled` field through an endpoint which accepts a pairingEnabled boolean query parameter
PUT api/users/{id}?pairingEnabled=
Request:
Path param: User id
Query param: pairingEnabled boolean
Response:
200 OK w/ updated User entity
404 Not Found

Pros:
- No confusion as to what field of User should be updated in the database

Cons:
- We don't have a generic update user endpoint

**Notes**
Need input from User and Pairing teams

### PR DESCRIPTION
**Pull request by uoa-ypec413**
## Description
Adds an endpoint to disable/enable pairing for a user so that a user can opt in to being paired with a buddy.
Changes:
- An endpoint in `UserController`:
PUT api/users/{id}?pairingEnabled=
Request:
Path param: User id
Query param: pairingEnabled boolean
Response:
200 OK w/ updated User entity
404 Not Found
- Added `updatePairingEnabled` to `UserService`
- Added a custom update query  `updatePairingEnabled` to UserRepository, to allow partial updating of the User object with only the `pairingEnabled` field
- Changed the id type from UUID to Long for the  `User` and `Buddies` entities, as using a UUID as the Primary Key is bad practice and complex

Fixes #34

## How has this been tested?
- Added a unit test for both the controller and service
- Created a new integration test for `UserController`

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/main/java/com/team701/buddymatcher/controllers/users/UserController.java
Comment (ff-dev-45) [inline src/main/java/com/team701/buddymatcher/controllers/users/UserController.java:35]: Nit: could this be changed to a more descriptive method name?

Comment (uoa-ypec413) [reply src/main/java/com/team701/buddymatcher/controllers/users/UserController.java:35]: Will do :)

File: src/main/java/com/team701/buddymatcher/controllers/users/UserController.java
Comment (shreytailor) [inline src/main/java/com/team701/buddymatcher/controllers/users/UserController.java:48]: After today's meeting and how we decided the authenticated endpoints will work, the `id` path parameter won't be required right? Since you'll be able to retrieve the `id` of the user through the  [middleware](https://github.com/SE701-T1/backend/issues/57)

What are your thoughts about this?

Comment (uoa-ypec413) [reply src/main/java/com/team701/buddymatcher/controllers/users/UserController.java:48]: Yeah that's the plan. 

I was wanting to just merge this now, and then add a seperate issue to refactor both this endpoint and the retrieveUserById endpoint once we implement authorisation.

I can also block this PR until we get authorisation merged. 

What do you think is best?

Comment (shreytailor) [reply src/main/java/com/team701/buddymatcher/controllers/users/UserController.java:48]: Yeah good point actually! We can merge it for now because the endpoint is technically fully functional, and we can change its implementation later once we complete the Auth middleware :)

**Review by ff-dev-45 — APPROVED** [review]
LGTM

File: src/main/java/com/team701/buddymatcher/dtos/pairing/AddBuddyDTO.java
Comment (uoa-ypec413) [inline src/main/java/com/team701/buddymatcher/dtos/pairing/AddBuddyDTO.java:7]: These pairing DTOs were refactored after changing the User id from UUID to Long. The integration test began failing once I rebased off the main repository.

File: src/test/java/com/team701/buddymatcher/controllers/pairing/PairingControllerIntegrationTest.java
Comment (uoa-ypec413) [inline src/test/java/com/team701/buddymatcher/controllers/pairing/PairingControllerIntegrationTest.java:60]: This test was failing due to the additional "Successfully". PairingController does not contain the word "Successfully"

**Review by ff-dev-45 — APPROVED** [review]
LGTM, good work on doing all of that!!

File: src/main/java/com/team701/buddymatcher/dtos/pairing/AddBuddyDTO.java
Comment (ff-dev-45) [reply src/main/java/com/team701/buddymatcher/dtos/pairing/AddBuddyDTO.java:7]: Good work for refactoring those!

**Pull request merged**
